pkg/oci/load_balancer.go

```diff
@@ -469,38 +431,36 @@ func (cp *CloudProvider) updateBackendSet(ctx context.Context, lbOCID string, ac
 	return nil
 }
 
-func (cp *CloudProvider) updateListener(ctx context.Context, lbOCID string, action *ListenerAction, backendPort int, healthCheckPort int, lbSubnets, nodeSubnets []*core.Subnet, sourceCIDRs []string) error {
-	var (
-		workRequestID string
-		err           error
-		l             = action.Listener
-		listenerPort  = *l.Port
-	)
+func (cp *CloudProvider) updateListener(ctx context.Context, lbID string, action *ListenerAction, ports portSpec, lbSubnets, nodeSubnets []*core.Subnet, sourceCIDRs []string) error {
+	var workRequestID string
+	var err error
+	l := action.Listener
```

Comment:
I'd avoid using variable names that are this ambiguous in this context. Would prefer for clarity 

listener := action.Listener